### PR TITLE
adds NativeCmds `cd` and `pwd`

### PR DESCRIPTION
**Pull request by audibleblink**
### Pull Request (PR) Checklist
- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.MD) doc
- [x] PR is from **a topic/feature/bugfix branch** off the **dev branch** (right side)
- [x] PR is against the **dev branch** (left side)
- [x] Merlin compiles without errors
- [x] Passes linting checks and unit tests
- [x] Updated [CHANGELOG](../CHANGELOG.MD)
- [ ] Updated README documentation (if applicable)

### Change Type
- [x] Addition
- [ ] Bugfix
- [ ] Modification
- [ ] Removal
- [ ] Security

### Description

Adds new `NativeCmd`s (`cd` & `pwd`) to agent and server.

### CODE REVIEW

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -897,10 +897,85 @@ func (a *Agent) statusCheckIn(host string, client *http.Client) {
...
+
+				c := messages.CmdResults{
+					Job:    p.Job,
+					Stdout: stdout,
+					Stderr: se,
+				}
+
+				k, err := json.Marshal(c)
+				if err != nil {
+					panic(err)
```
Comment (Ne0nd0g) [inline pkg/agent/agent.go]: I must have missed this on the last NativeCMD PR. For the sake of availability, I would like the agent to silently continue as opposed to exiting. I'm open to feedback though.

File: pkg/agent/agent.go [hunk 2]
```diff
@@ -897,10 +897,85 @@ func (a *Agent) statusCheckIn(host string, client *http.Client) {
...
+					Type:    "CmdResults",
+					Payload: (*json.RawMessage)(&k),
+					Padding: core.RandStringBytesMaskImprSrc(a.PaddingMax),
+				}
+				b2 := new(bytes.Buffer)
+				json.NewEncoder(b2).Encode(g)
+				if a.Verbose {
+					message("note", fmt.Sprintf("Sending response to server: %s", stdout))
+				}
+				resp2, _ := client.Post(host, "application/json; charset=utf-8", b2)
```
Comment (Ne0nd0g) [inline pkg/agent/agent.go]: Receive and handle the error.

File: pkg/agent/agent.go [hunk 3]
```diff
@@ -897,10 +897,85 @@ func (a *Agent) statusCheckIn(host string, client *http.Client) {
...
-			}
-		default:
-			if a.Verbose {
-				message("warn", fmt.Sprintf("Received unrecognized message type: %s", j.Type))
+			case "cd":
+				var se string
+				var stdout string
+				err := os.Chdir(p.Args)
+				if err != nil {
+					se = err.Error()
```
Comment (Ne0nd0g) [inline pkg/agent/agent.go]: Please add a CLI message if verbose is on.

File: pkg/agent/agent.go [hunk 4]
```diff
@@ -897,10 +897,85 @@ func (a *Agent) statusCheckIn(host string, client *http.Client) {
...
+				if resp2.StatusCode != 200 {
+					if a.Verbose {
+						message("warn", fmt.Sprintf("Message error from server. HTTP Status code: %d", resp2.StatusCode))
+					}
+				}
+			case "pwd":
+				var se string
+				dir, err := os.Getwd()
+				if err != nil {
+					se = err.Error()
```
Comment (Ne0nd0g) [inline pkg/agent/agent.go]: Please add a CLI message if verbose is on.

File: pkg/agent/agent.go [hunk 5]
```diff
@@ -897,10 +897,85 @@ func (a *Agent) statusCheckIn(host string, client *http.Client) {
...
+
+				c := messages.CmdResults{
+					Job:    p.Job,
+					Stdout: dir,
+					Stderr: se,
+				}
+
+				k, err := json.Marshal(c)
+				if err != nil {
+					panic(err)
```
Comment (Ne0nd0g) [inline pkg/agent/agent.go]: Same comments as before regarding a panic.

File: pkg/agent/agent.go [hunk 6]
```diff
@@ -897,10 +897,85 @@ func (a *Agent) statusCheckIn(host string, client *http.Client) {
...
+					Type:    "CmdResults",
+					Payload: (*json.RawMessage)(&k),
+					Padding: core.RandStringBytesMaskImprSrc(a.PaddingMax),
+				}
+				b2 := new(bytes.Buffer)
+				json.NewEncoder(b2).Encode(g)
+				if a.Verbose {
+					message("note", fmt.Sprintf("Sending response to server: %s", dir))
+				}
+				resp2, _ := client.Post(host, "application/json; charset=utf-8", b2)
```
Comment (Ne0nd0g) [inline pkg/agent/agent.go]: Receive and handle the error.

**Comment by audibleblink** [comment]
Happy to contribute! I've made the recommended changes.

**Pull request merged**